Source/WebCore/platform/Scrollbar.cpp

```diff
@@ -286,7 +286,7 @@ void Scrollbar::moveThumb(int pos, bool draggingDocument)
             destinationPosition = std::min(destinationPosition + delta, maximum());
         else if (delta < 0)
             destinationPosition = std::max(destinationPosition + delta, 0);
-        m_scrollableArea.scrollToOffsetWithoutAnimation(m_orientation, destinationPosition);
+        m_scrollableArea.scrollToPositionWithoutAnimation(m_orientation, destinationPosition);
```

Comment:
This is a nice complement to https://commits.webkit.org/272466@main which fixes dragging the thumb with the Option key held down (see 74496@main).

But let's fixing it by keeping the call to `scrollToOffsetWithoutAnimation()`, and just doing the math above using the offset, not the position.

Reply:
Also a test would be nice (see my commit for an example).

Reply:
So several options here are:
- Make ScrollAnimator::scrollOffset() public and call it (and maybe rename it to currentOffset()) instead of currentPosition(). This does some kind of rounding. Then convert everything to offsets.
- Same but use the ScrollArea::scrollOffsetFromPosition() instance method, which does int conversion.
- Same but duplicate the currentPosition to currentOffset math in `ScrollableArea::scrollToOffsetWithoutAnimation`, which uses floats. (Note it'll be more awkward here than inside ScrollableArea, because we'll be pulling multiple numbers up out of ScrollableArea to do math on them before shoving them back down into ScrollableArea for the scrollTo call.)
- Keep the current approach of working in positions, which is ultimately what we're doing anyway, instead of converting to offsets and back.